### PR TITLE
Server side signup validation

### PR DESCRIPTION
**Pull request by KajenKirubah**

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -19,14 +19,14 @@ app.use(urlencoded({ extended: false }));
 app.use(cookieParser());
 app.use(express.static(join(__dirname, "public")));
 
-mongoose.connect(process.env.MONGODB_URI, {
+mongoose.connect('mongodb://team-maple:map1e-password@ds133137.mlab.com:33137/team-maple', {
```
Comment (skassam21) [inline server/app.js:22]: Use process.env for this and don't commit this url

File: server/app.js [hunk 2]
```diff
@@ -38,11 +38,12 @@ app.use(function(req, res, next) {
 // error handler
 app.use(function(err, req, res, next) {
   // set locals, only providing error in development
+  console.log(err);
```
Comment (skassam21) [inline server/app.js:41]: Ideally use a logging library - add a ticket in the icebox to do that later

File: server/controllers/auth.js [hunk 3]
```diff
@@ -1,8 +1,20 @@
...
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    return res
+      .status(422)
+      .json({ message: "validation failed", errorData: errors.array() });
+    // const error = new Error('Validation failed');
+    // error.statusCode = 422;
+    // error.data = error.array();
+
+    // throw error;
```
Comment (skassam21) [inline server/controllers/auth.js:16]: avoid commented code like this

File: server/routes/auth.js [hunk 4]
```diff
@@ -1,9 +1,54 @@
...
+          }
+        });
+      })
+      .normalizeEmail(),
+    body("password")
+      .trim()
+      .isLength({ min: 8 })
+  ],
+  authController.login
+);
```
Comment (skassam21) [inline server/routes/auth.js:52]: I don't love how the code for validation is in this file

File: server/routes/auth.js [hunk 5]
```diff
@@ -1,9 +1,54 @@
...
+          if (exEmail) {
+            return Promise.reject("Email already exists!");
+          }
+        });
+      })
+      .normalizeEmail(),
+    body("password")
+      .trim()
+      .isLength({ min: 8 })
+  ],
```
Comment (skassam21) [inline server/routes/auth.js:50]: don't need this validation for login

**Comment by KajenKirubah** [comment]
older branch, so im closing it

**Pull request closed without merging**